src/fswAlgorithms/pointCloudProcessing/initializeICP/_UnitTest/test_initICP.py

```diff
@@ -0,0 +1,183 @@
+    ephemeris_input_msg_buffer.sigma_BN = [0.1, 0., 1]  # use v_C1_N instead of v_BN_N for accurate unit test
+    ephemeris_input_msg_buffer.timeTag = 1 * 1E9
+    ephemeris_input_msg = messaging.EphemerisMsg().write(ephemeris_input_msg_buffer)
+    module.ephemerisInMsg.subscribeTo(ephemeris_input_msg)
+
+    camera_input_msg_buffer = messaging.CameraConfigMsgPayload()
+    camera_input_msg_buffer.cameraID = 1
+    camera_input_msg_buffer.sigma_CB = [0.1, 0.2, 0.3]
+    camera_input_msg = messaging.CameraConfigMsg().write(camera_input_msg_buffer)
+    module.cameraConfigInMsg.subscribeTo(camera_input_msg)
+
+    input_points = np.array([[1, 1, 1], [2, 2, 2], [5, 1, 3], [0, 2, 0], [0, 0, 1], [0, 0, -1]])
+    pointcloud_input_msg_buffer = messaging.PointCloudMsgPayload()
+    pointcloud_input_msg_buffer.points = input_points.flatten().tolist()
+    pointcloud_input_msg_buffer.numberOfPoints = len(input_points)
+    pointcloud_input_msg_buffer.timeTag = 1
+    pointcloud_input_msg_buffer.valid = validCloud
+    pointcloud_input_msg = messaging.PointCloudMsg().write(pointcloud_input_msg_buffer)
+    module.inputMeasuredPointCloud.subscribeTo(pointcloud_input_msg)
+
+    icp_input_msg_buffer = messaging.SICPMsgPayload()
+    icp_input_msg_buffer.rotationMatrix = [1, 0, 0, 0, 1, 0, 0, 0, 1]
+    icp_input_msg_buffer.translation = [1, 0, 0]
+    icp_input_msg_buffer.scaleFactor = [1.05]
+    icp_input_msg_buffer.numberOfIteration = 1
+    icp_input_msg_buffer.valid = validICP
+    icp_input_msg_buffer.timeTag = 500
+    icp_input_msg = messaging.SICPMsg().write(icp_input_msg_buffer)
+    module.inputSICPData.subscribeTo(icp_input_msg)
+
+    icp_msg_log = module.initializeSICPMsg.recorder()
+    unit_test_sim.AddModelToTask(unit_task_name, icp_msg_log)
+    cloud_msg_log = module.measuredPointCloud.recorder()
+    unit_test_sim.AddModelToTask(unit_task_name, cloud_msg_log)
+
+    unit_test_sim.InitializeSimulation()
+    unit_test_sim.ConfigureStopTime(process_rate)
+    unit_test_sim.ExecuteSimulation()
+
+    output_iterations = icp_msg_log.numberOfIteration[-1] + 1
+    output_Ss = icp_msg_log.scaleFactor[-1][:output_iterations]
+    output_Rs = icp_msg_log.rotationMatrix[-1][:9*output_iterations]
+    output_Ts = icp_msg_log.translation[-1][:3*output_iterations]
+
+    output_cloud_valid = cloud_msg_log.valid[-1]
+    output_cloud_timetag = cloud_msg_log.timeTag[-1]
+    output_cloud_numberpoints = cloud_msg_log.numberOfPoints[-1]
+    output_pointcloud = cloud_msg_log.points[-1][:3*output_cloud_numberpoints]
+
+    point_cloud_module = np.array(output_pointcloud).reshape([output_cloud_numberpoints, 3])
+
+    accuracy = 1E-10
+    if module.normalizeMeasuredCloud:
+        avg = np.mean(np.linalg.norm(input_points, axis=1))
+    else:
+        avg = 1
+    if validCloud and validICP:
+        np.testing.assert_allclose(point_cloud_module,
+                                   input_points/avg,
+                                   atol=accuracy,
```

Comment:
:large_orange_diamond: If you don't want an `rtol` or `atol` check then the one you don't want should be set to 0 otherwise then `rtol` defaults to 1E-7 and `atol` to 0 (probably fine).

Reply:
Added for clarity